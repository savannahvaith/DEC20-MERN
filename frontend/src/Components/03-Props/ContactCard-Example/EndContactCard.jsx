'use strict';

const EndContactCard = ({img,name,phone,email}) => {
    return (
        <div className="contact-card">
            <img src={img} />
            <h3>{name}</h3>
            <p>Phone: {phone}</p>
            <p>Email: {email}</p>
        </div>
    )
}
export default EndContactCard; 