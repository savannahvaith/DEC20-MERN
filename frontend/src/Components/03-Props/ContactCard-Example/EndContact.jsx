'use strict';

import EndContactCard from "./EndContactCard";

const EndContact = () => {
    return (
        <div className="contacts">
            <EndContactCard 
                name="Harry Pawter" 
                phone="123234345" 
                email="harry.pawter@gmail.com" 
                img="https://placedog.net/300/200?id=43"
            />
            <EndContactCard 
                name="Jimmy Chew" 
                phone="234345456" 
                email="mr.jimmy@gmail.com" 
                img="https://placedog.net/400/200?id=5"/>
            <EndContactCard 
                name="Bark Wahlberg"
                img="https://placedog.net/300/200?id=66"
                phone="(212) 555-3456"
                email="bark@facebook.woof"
            />
            <EndContactCard 
                name="Indiana Bones"
                img="https://placedog.net/300/240?id=3"
                phone="(212) 555-5678"
                email="indiana.bones@hotmail.com" 
            />
        </div>
    )
}

export default EndContact; 