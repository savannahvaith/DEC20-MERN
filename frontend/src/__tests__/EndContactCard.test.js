import EndContactCard from '../Components/03-Props/ContactCard-Example/EndContactCard';
import {create} from 'react-test-renderer';
import EndContact from '../Components/03-Props/ContactCard-Example/EndContact';

describe(`End Contact Card Testing with Props`, () => {

    const props = {
        img: `https://placedog.net/300/200?id=43`,
        name: `Harry Pawter`, 
        phone: `01234567890`, 
        email: "harry@therealone.com"
    }

    let componentToTest;

    beforeEach( () => {
        const TestInstance = create(<EndContactCard {...props}/>);
        componentToTest = TestInstance.root; 
        // console.log(componentToTest);
    });

    it(`Should render a h3 with the name of the puppy`, () => {
        const h3 = componentToTest.findByType(`h3`);
        // console.log(img)
        expect(h3.children).toEqual([props.name]);
    });

    it(`Should render the phone in the first p tag`, () => {
        const firstP = componentToTest.findAllByType(`p`)[0];
        console.log(firstP.children[1])
        expect(firstP.children[1]).toContain([props.phone]);
    });

    it(`Should match the snapshot!!`, () => {
        const instance = create(<EndContactCard {...props}/>).toJSON();
        expect(instance).toMatchSnapshot();
    });

    const childComp = jest.mock(`../Components/03-Props/ContactCard-Example/EndContactCard`, () => {
        return function EndContact(props){
            return(<p>Mocked components!</p>);
        }
    });

    it(`should match this:`, () => {
        const instance = create(<EndContact/>).toJSON();
        expect(instance).toReturnWith(childComp);
    });

});