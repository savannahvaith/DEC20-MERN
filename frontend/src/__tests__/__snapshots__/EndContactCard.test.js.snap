// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`End Contact Card Testing with Props Should match the snapshot!! 1`] = `
Array [
  <img
    src="https://placedog.net/300/200?id=43"
  />,
  <h3>
    Harry Pawter
  </h3>,
  <p>
    Phone: 
    01234567890
  </p>,
  <p>
    Email: 
    harry@therealone.com
  </p>,
]
`;
